packages/mermaid/src/diagrams/gantt/ganttDb.spec.ts

```diff
@@ -432,4 +432,10 @@ describe('when using the ganttDb', function () {
     ganttDb.setTodayMarker(expected);
     expect(ganttDb.getTodayMarker()).toEqual(expected);
   });
+
+  it('should reject dates with ridiculous years', function () {
+    ganttDb.setDateFormat('YYYYMMDD');
+    ganttDb.addTask('test1', 'id1,202304,1d');
+    expect(() => ganttDb.getTasks()).toThrowError('Invalid date:202304');
```

Comment:
Note that this error is dependent on the behavior of V8/JavaScriptCore.  SpiderMonkey's `Date` constructor will throw its own `Invalid Date` error, which means that the logic introduced in this PR will never even execute, and the error thrown will not match this string.  But I'm assuming that these tests will only ever be run in Node/V8--let me know if I'm mistaken!